html/index.html

```diff
@@ -0,0 +1,332 @@
+                <th scope="row">CountryEU</th>
+                <td>{{ .CountryEU }}</td>
+              </tr>
+              {{ end }} {{ if .RegionName }}
+              <tr>
+                <th scope="row">RegionName</th>
+                <td>{{ .RegionName }}</td>
+              </tr>
+              {{ end }} {{ if .RegionCode }}
+              <tr>
+                <th scope="row">RegionCode</th>
+                <td>{{ .RegionCode }}</td>
+              </tr>
+              {{ end }} {{ if .MetroCode }}
+              <tr>
+                <th scope="row">MetroCode</th>
+                <td>{{ .MetroCode }}</td>
+              </tr>
+              {{ end }} {{ if .PostalCode }}
+              <tr>
+                <th scope="row">PostalCode</th>
+                <td>{{ .PostalCode }}</td>
+              </tr>
+              {{ end }} {{ if .City }}
+              <tr>
+                <th scope="row">City</th>
+                <td>{{ .City }}</td>
+              </tr>
+              {{ end }} {{ if .Latitude }}
+              <tr>
+                <th scope="row">Latitude</th>
+                <td>{{ .Latitude }}</td>
+              </tr>
+              {{ end }} {{ if .Longitude }}
+              <tr>
+                <th scope="row">Longitude</th>
+                <td>{{ .Longitude }}</td>
+              </tr>
+              {{ end }} {{ if .Timezone }}
+              <tr>
+                <th scope="row">Timezone</th>
+                <td>{{ .Timezone }}</td>
+              </tr>
+              {{ end }} {{ if .ASN }}
+              <tr>
+                <th scope="row">ASN</th>
+                <td>{{ .ASN }}</td>
+              </tr>
+              {{ end }} {{ if .ASNOrg }}
+              <tr>
+                <th scope="row">ASNOrg</th>
+                <td>{{ .ASNOrg }}</td>
+              </tr>
+              {{ end }} {{ if .Hostname }}
+              <tr>
+                <th scope="row">Hostname</th>
+                <td>{{ .Hostname }}</td>
+              </tr>
+              {{ end }} {{ if .UserAgent }} {{ if .UserAgent.Comment }}
+              <tr>
+                <th scope="row">User Agent</th>
+                <td>{{ .UserAgent.Product }}/{{ .UserAgent.Version }}</td>
+              </tr>
+              {{ end }} {{ if .UserAgent.Comment }}
+              <tr>
+                <th scope="row">Comment</th>
+                <td>{{ .UserAgent.Comment }}</td>
+              </tr>
+              {{ end }} {{ if .UserAgent.RawValue }}
+              <tr>
+                <th scope="row">Raw Value</th>
+                <td>{{ .UserAgent.RawValue }}</td>
+              </tr>
+              {{ end }} {{ end }}
+            </table>
+            {{ if .Country }}
+            <p>
+              This information is provided from the GeoLite2 database created by
+              MaxMind, available from
+              <a href="https://www.maxmind.com">www.maxmind.com</a>
+            </p>
+            {{ end }} {{ if .City }}
```

Comment:
Technically you need to check latitude and longitude instead of city, I guess, unless you know that those parameters always exist if city is known?

Reply:
√